modules/Rock.cmake

```diff
@@ -49,6 +49,31 @@ function(rock_add_compiler_flag_if_it_exists FLAG)
     endif()
 endfunction()
 
+macro(ADD_TARGET_PROPERTIES _target _name _properties)
+    set(_properties ${ARGV})
+    list(REMOVE_AT _properties 0)
+    list(REMOVE_AT _properties 0)
+    get_target_property(_old_properties ${_target} ${_name})
+    if(NOT _old_properties)
+        set(_old_properties)
+    else(NOT _old_properties)
```

Comment:
repeating the arguments to CMake constructs has not been needed for a very long time ... and is deprecated. Don't do it, e.g.

```
if (xxxxx)
else()
endif()
```